module-system/module-implementations/sov-accounts/src/lib.rs

```diff
@@ -82,54 +83,141 @@ impl<C: Context> sov_modules_api::Module for Accounts<C> {
+
+        fn arbitrary_with(_args: Self::Parameters) -> Self::Strategy {
+            any::<Vec<C::PrivateKey>>()
+                .prop_map(|keys| AccountConfig {
+                    pub_keys: keys.into_iter().map(|k| k.pub_key()).collect(),
+                })
+                .boxed()
+        }
+    }
 
-        accounts
-            .genesis(&config, working_set)
-            .map_err(|_| arbitrary::Error::IncorrectFormat)?;
+    impl<'a, C> Accounts<C>
+    where
+        C: Context,
+        C::Address: Arbitrary<'a>,
+        C::PublicKey: Arbitrary<'a>,
+    {
+        /// Creates an arbitrary set of accounts and stores it under `working_set`.
+        pub fn arbitrary_workset(
+            u: &mut Unstructured<'a>,
+            working_set: &mut WorkingSet<C>,
+        ) -> arbitrary::Result<Self> {
+            let config: AccountConfig<C> = u.arbitrary()?;
+            let accounts = Accounts::default();
+
+            accounts
+                .genesis(&config, working_set)
+                .map_err(|_| arbitrary::Error::IncorrectFormat)?;
+
+            Ok(accounts)
+        }
+    }
 
-        Ok(accounts)
+    impl<C> Accounts<C>
+    where
+        C: Context,
+        C::Address: proptest::arbitrary::Arbitrary,
+        C::PrivateKey: proptest::arbitrary::Arbitrary,
+    {
+        /// Creates an arbitrary set of accounts and stores it under `working_set`.
+        ///
+        /// We take the `WorkingSet` as `Arc<Mutex<_>>` so the strategy can freely implement
+        /// parallelism while preserving interior mutability safety. The return is a reference to a
+        /// promise of a strategy; hence, the `WorkingSet` lifetime must be locked to its lifetime.
+        pub fn arbitrary_proptest_workset(
+            working_set: Arc<Mutex<WorkingSet<C>>>,
+        ) -> impl Strategy<Value = Self> {
+            any::<(C::Address, AccountConfig<C>, Prefix, Prefix)>()
+                .prop_perturb(
+                    move |(address, config, prefix_keys, prefix_accounts), mut rng| {
+                        let mut working_set_lock =
+                            working_set.lock().expect("working set lock is poisoned");
+                        let working_set = &mut *working_set_lock;
+                        let public_keys = StateMap::new(prefix_keys);
+                        let accounts = StateMap::new(prefix_accounts);
+
+                        for public in config.pub_keys {
+                            let mut address = [0u8; 32];
+                            rng.fill_bytes(&mut address);
+
+                            let addr = C::Address::from(address);
+                            let nonce = rng.next_u64();
+                            let account: Account<C> = Account { addr, nonce };
+
+                            public_keys.set(&account.addr, &public, working_set);
+                            accounts.set(&public, &account, working_set);
+                        }
+
+                        Accounts {
```

Comment:
The ` Accounst.accounts, Accounts.public_keys` prefixes are set by a macro and are always the same.  Why do we fuzz the `Accounts` structure? I thought we will only fuzz the CallMessages.

Reply:
Makes sense, updated to `Accounts::default`. As the previous comment, this is just a convenience function available. Do you think its best to remove it?

Reply:
Ahh ok now I got it, but is it going to be used somwhere? If it is not used the let's remove it. Btw should move the fuzz tests in the modules?

Reply:
Makes sense; moved to `fuzz`